spring-cloud-dataflow-docs/src/main/asciidoc/configuration-database.adoc

```diff
@@ -0,0 +1,65 @@
+
+[[configuration-database-overview]]
+A relational database is used to store stream and task definitions as well as the state of executed tasks.
+Spring Cloud Data Flow provides schemas for *MariaDB*, *MySQL*, *Oracle*, *PostgreSQL*, *Db2*, *SQL Server*, and *H2*. The schema is automatically created when the server starts.
+
+NOTE: The JDBC drivers for *MariaDB*, *MySQL* (via the _MariaDB_ driver), *PostgreSQL*, *SQL Server* are available without additional configuration. To use any other database you need to put the corresponding JDBC driver jar on the classpath of the server as described <<#add-custom-driver,here>>.
+
+To configure a database the following properties must be set:
+
+* `spring.datasource.url`
+* `spring.datasource.username`
+* `spring.datasource.password`
+* `spring.datasource.driver-class-name`
+
+The `username` and `password` are the same regardless of the database. However, the `url` and `driver-class-name` vary per database as follows.
+
+[frame="none"]
+[cols="a,a,a,a"]
+[cols="10%,30%,20%,10%"]
+|===
+|[.small]#Database#|[.small]#spring.datasource.url#|[.small]#spring.datasource.driver-class-name#|[.small]#Driver included#
+
+|[.small]#MariaDB 10.4+#
+|[.small]#jdbc:mariadb://${db-hostname}:${db-port}/${db-name}#
+|[.small]#org.mariadb.jdbc.Driver#
+|[.small]#Yes#
+
+|[.small]#MySQL 5.7#
+|[.small]#jdbc:mysql://${db-hostname}:${db-port}/${db-name}?permitMysqlScheme#
+|[.small]#org.mariadb.jdbc.Driver#
+|[.small]#Yes#
+
+|[.small]#MySQL 8.0+#
+|[.small]#jdbc:mariadb://${db-hostname}:${db-port}/${db-name}?allowPublicKeyRetrieval=true&useSSL=false&autoReconnect=true&permitMysqlScheme#{empty}footnote:[SSL is disabled in this example, adjust accordingly for your environment and requirements]
+|[.small]#org.mariadb.jdbc.Driver#
+|[.small]#Yes#
+
+|[.small]#PostgresSQL#
+|[.small]#jdbc:postgres://${db-hostname}:${db-port}/${db-name}#
+|[.small]#org.postgresql.Driver#
+|[.small]#Yes#
+
+|[.small]#SQL Server#
+|[.small]#jdbc:sqlserver://${db-hostname}:${db-port};databasename=${db-name}&encrypt=false#
+|[.small]#com.microsoft.sqlserver.jdbc.SQLServerDriver#
+|[.small]#Yes#
+
+|[.small]#DB2#
+|[.small]#jdbc:db2://${db-hostname}:${db-port}/{db-name}#
+|[.small]#com.ibm.db2.jcc.DB2Driver#
+|[.small]#No#
+
+|[.small]#Oracle#
+|[.small]#jdbc:oracle:thin:@${db-hostname}:${db-port}/{db-name}#
+|[.small]#oracle.jdbc.OracleDriver#
+|[.small]#No#
+|===
+
+==== H2
+When no other database is configured and the *H2* driver has been added to the server classpath then
```

Comment:
Noob question, Is there a case where we do not include the H2 driver?   If they don't specify a datasource dataflow defaults to H2.

Reply:
So....

The `H2` config was split into its own config and made optional in #4802 as it was lighting up CVE scans. 

We quickly realized the missing dep. broke running locally w/ `java -jar` as well as many examples that do the same in the docs. Because of this I made the dependency non-optional [here](https://github.com/spring-cloud/spring-cloud-dataflow/commit/7525254469ed52fa75b86ccc5c814c9e0c99de97).

As of now the dependency is still provided. We need to decide what we are going to do in this area (provide it or not). This doc is written as if it will be removed. Lets leave it like that for now and we will re-address the docs when we make a decision on H2. 

Sound good @cppwfs ?

Reply:
I'm going to merge this and we can follow on w/ whatever we need to.